libbpf-rs/src/streams.rs

```diff
@@ -0,0 +1,105 @@
+use core::ffi::c_void;
+use std::io;
+
+use crate::libbpf_sys::bpf_prog_stream_read_opts;
+
+use crate::Error;
+use crate::Result;
+
+/// [`Stream`] instances.
+///
+/// `streams`s are BPF descriptors that provide a character output
+/// interface from the BPF program to userspace.
+#[derive(Debug)]
+pub struct Stream {
+    fd: i32,
+    stream_id: u32,
+    bytes_per_read: usize,
+}
+
+/// Default BPF stdout stream id.
+pub const BPF_STDOUT: u32 = 1;
+/// Default BPF stderr stream id.
+pub const BPF_STDERR: u32 = 2;
+
+/// Config options, currently empty and here for forward compatibility.
+struct StreamReadOpts {
+    sz: usize,
+}
+
+impl Stream {
+    /// Default read batch for when the user doesn't specify one.
+    const BYTES_PER_READ_DFL: usize = 4096;
+
+    /// Create a new Stream instance.
+    pub fn new(fd: i32, stream_id: u32) -> Self {
```

Comment:
So what is a "stream_id" and what is its purpose? It's unclear what relevance it has to the user. I don't see it mentioned in the only "documentation" that I can find on the topic, which is https://lwn.net/Articles/1026660/ (and its reincarnations)